### ISSUE TITLE
v2 - AuthnContextClassRef not included in SAML Request

### ISSUE THREAD

**Issue by holandaleo** [spring-projects/spring-security-saml#267]
Hi, I'm trying to get the SP sample working with SSO Circle. I was getting 500 error and I was able to debug the SAML response with contained:

```
<samlp:Response xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol" ID="s264a2..." InResponseTo="84cf7..." Version="2.0" IssueInstant="2018-07-13T18:39:10Z">
	<saml:Issuer xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion">https://idp.ssocircle.com</saml:Issuer>
	<samlp:Status xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol">
		<samlp:StatusCode xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol" Value="urn:oasis:names:tc:SAML:2.0:status:Requester">
			<samlp:StatusCode xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol" Value="urn:oasis:names:tc:SAML:2.0:status:NoAuthnContext">
			</samlp:StatusCode>
		</samlp:StatusCode>
	</samlp:Status>
</samlp:Response>

```

I did a bit of search on StackOverflow and found that including `AuthnContextClassRef` [would solve](https://stackoverflow.com/q/16353803) the problem.

I tried adding `AuthenticationContextClassReference.PASSWORD_PROTECTED_TRANSPORT` to the defaults of `AuthenticationRequest` and it works properly. Is there another way to fix this by adding some config property to application.yml or something? Is it ok if I push the changes I made to a separate branch created out of commit 009d145ead to share what I did? This would be my first ever contribution to a public project on github.

**Comment by fhanik** [comment]
@holandaleo yes, please fork the repository, and create a Pull request.

https://help.github.com/articles/creating-a-pull-request/

If you add a test you get bonus points :)

**Comment by fhanik** [comment]
@holandaleo please also look at https://github.com/spring-projects/spring-security-saml/pull/264

Maybe the solution is not to have anything at first, and then programmers can add any AuthenticationContextClassReference they want by extending the library

**Comment by holandaleo** [comment]
Yea. I pulled latest after #264 was merged and it fixes the issue I was having. However, when I was updating the test cases as you suggested, I realized, regardless of the defaults, `toXml()` and `fromXml()` are ignoring the `AuthnContextClassRef` when present. I'll prepare a pull request without my changes to `defaults()` but `OpenSamlImplementation` will still handle `AuthnContextClassRef` when present.
